spyder/workers/updates.py

```diff
@@ -225,16 +227,37 @@ def _progress_reporter(self, block_number, read_size, total_size):
     def _download_installer(self):
         """Donwload latest Spyder standalone installer executable."""
         tmpdir = tempfile.gettempdir()
-        is_full_installer = (is_module_installed('numpy') or
-                             is_module_installed('pandas'))
-        if os.name == 'nt':
-            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
-                                                else 'lite')
+
+        # If running on macOS unde Rosetta, platform.machine will not be
+        # correct so check for ARM64 in version instead. All other
+        # platforms will be x86_64.
+        mach = "arm64" if "ARM64" in platform.version() else "x86_64"
+
+        if parse(self.latest_release_version) >= parse("5"):
+            # conda-based installer name
+            if os.name == 'nt':
+                plat, ext = 'Windows', 'exe'
+            elif platform.system == 'Darwin':
+                plat, ext = 'macOS', 'pkg'
+            else:
+                plat, ext = 'Linux', 'sh'
+            name = f'Spyder_{plat}_{mach}.{ext}'
         else:
-            name = 'Spyder{}.dmg'.format('' if is_full_installer else '-Lite')
+            # 5.x installer name
+            is_full_installer = (is_module_installed('numpy') or
+                                 is_module_installed('pandas'))
+            if os.name == 'nt':
+                name = 'Spyder_64bit_{}.exe'.format(
+                    'full' if is_full_installer else 'lite'
+                )
+            else:
+                name = 'Spyder_{}{}.dmg'.format(
+                    mach, '' if is_full_installer else '_Lite'
```

Comment:
Revert the name to the previous for INtel

```suggestion
                    '' if mach == ''x86_64' else mach, '' if is_full_installer else '_Lite'
```